### DISCUSSION TITLE
Authentication support for web app

### DISCUSSION THREAD

**Discussion by m3nu** [ananthakumaran/paisa#114] [Ideas]
Was just trying to add Paisa as supported app for [pikapods.com](http://pikapods.com) since some users asked for it. Everything worked well, but I couldn't find a way to add authentication. I'm assuming that's not implemented yet?

I'll look into adding basic auth at the server level, but having this in the app would be useful too. If the Docker/web app deployment method is a focus.

**Comment by ananthakumaran** [comment]
Thanks for looking into Paisa. Auth is not yet implemented. It has been requested a few times, but I haven't figured out the details yet. I need to think about the implementation since paisa is available as two variants (desktop app, web UI), and what happens if the user forgets the password, does it protect the ledger file with encryption, etc. 

I checked the Pikapods documentation and couldn't find anything about app authentication, is it safe to assume Pikapod will just proxy the connection to paisa port? and won't do anything else. I am asking because I had checked Sandstorm in the past and it had specific requirements about how to handle auth.

**Comment by m3nu** [comment]
Usually each app will implement authentication. Some only use basic auth that's set via env var. That's the simplest option probably.

Yes, PikaPods will simply run the Docker image and proxy to the port. There are no specific requirements and we usually use the unmodified official image.

Would be nice to add Paisa, as we already have Actual, which is another popular finance app people like. Right now is also a good time for this niche with the shutdown of Mint.

Thanks for the consideration!

**Reply by ananthakumaran** [reply]
Authentication is supported on the latest version (0.6.1)

**Reply by m3nu** [reply]
Awesome! That was fast. 👏

**Reply by Mozenraath** [reply]
Suggestion- can you add confirm password as step for this. Due to my faulty keyboard, I mistype the password and lost the access to complete setup on PikaPods. Had to setup everything again.

**Reply by ananthakumaran** [reply]
>  I mistype the password and lost the access to complete setup on PikaPods. Had to setup everything again.

it's possible to recover if you can edit the config file directly, you just need to remove the username/password entry under user_accounts. Pikapods allows file edit via sftp https://docs.pikapods.com/manage/files

Let me think about adding confirm password field as well.

**Comment by m3nu** [comment]
Now available on PikaPods: 🍾 https://www.pikapods.com/apps#finance

As a next step we can also look into doing a revenue sharing agreement and adding PikaPods as additional install option for users that don't run their own server.

**Reply by ananthakumaran** [reply]
This looks great. Could you shoot me an email (available on GitHub profile) with more info? By the way, I am trying out new logos as I type this, since the current one doesn't look nice on a black background, I am experimenting with a colored one.

**Reply by ananthakumaran** [reply]
@m3nu the new logo is available under brands folder https://github.com/ananthakumaran/paisa/tree/master/brand

**Reply by m3nu** [reply]
Looks good. Updating now.

**Comment by swarajsomala** [comment]
@m3nu , Thanks for sharing the link(https://www.pikapods.com/apps#finance), I don't know much about the servers and web technologies so I want to deploy my ledger details somewhere and run paisa on top of that. I think I can achieve that using _pikapods_, I already added the data using paisa editor but is it the only option to add the new transactions or any other way to connect from local machine like ssh or using docker and add transactions?

**Reply by m3nu** [reply]
You can upload your files via SFTP: https://docs.pikapods.com/manage/files

I quickly tested it and saw all the DB and ledger files. But do check if it works as expected. We can still do adjustments on the deployment.
